TodoApp/test/TodoApp.Api.Tests/Controllers/ItemsControllerTests.test.cs

```diff
@@ -113,8 +113,7 @@ public void SetUp()
             var updateItem = new Item {Id = id, Text = "Make a coffee"};
 
             _repository.UpdateAsync(id,
-                               Arg.Is<Item>(item => item.Id.ToString() == "c5cc89a0-ab8d-4328-9000-3da679ec02d3"
-                                                    && item.Text == "Make a coffee"))
+                               Arg.Is<Item>(item => item.Equals(updateItem) ))
```

Comment:
Unless I'm mistaken, you are using reference comparison here, meaning your `updateItem` is passed through the controller to the repository. This is a bit fragile, but I would rather discuss it in person than write an essay.